### PR TITLE
Fargate/Compress video with ECS

### PR DESCRIPTION
**Pull request by Awsaf780**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
- Compress video with ECS Task

## What was done
- Add ECS execution role and policy
- Change default deployment bucket
- Create Task Definition resource in YAML
- Build docker image for compression logic

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: awsaf-serverless-engine/src/handler.js [hunk 1]
```diff
@@ -173,19 +174,59 @@ async function queryTargetActivityDDB(keyConditionExpression, expressionAttribut
...
+        }
+      ]
+    }
+  };
+
+  try {
+    await ecs.runTask(ecsParams).promise();
+    return sendResponseToClient(200, "Task Ran Successfully");
+  } catch (error) {
+    return sendResponseToClient(400, error);
```
Comment (mhazad) [inline awsaf-serverless-engine/src/handler.js:213]: Please explain with proper logic why do you think the error code should be 400?

File: fargate/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "name": "fargate",
+    "version": "1.0.0",
+    "description": "",
+    "main": "compress.js",
+    "scripts": {
+      "test": "echo \"Error: no test specified\" && exit 1",
+      "compress": "run-func compress.js initiateCompression"
```
Comment (mhazad) [inline fargate/package.json:8]: Why do we need run-func here? Do we really need a third-party library to execute our js file?

File: fargate/compress.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            ["-i", `${inputFileS3Url}`, "-f", "h264", "-s", "852x480", "-y", tempFilePath],
+            { stdio: "inherit" });
+
+        const outputFile = readFileSync(tempFilePath);
+
+        unlinkSync(tempFilePath);
+
+        return putConvertedVideoToBucket(convertedFilename, outputFile);
+    } catch (error) {
+        return sendResponseToClient(400, error);
```
Comment (mhazad) [inline fargate/compress.js:44]: Please explain with proper logic why do you think the error code should be 400?

File: fargate/compress.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    try {
+        const params = new PutObjectCommand({
+            Bucket: process.env.BUCKET_NAME,
+            Key: convertedFilename,
+            Body: outputFile
+        });
+        await s3Client.send(params);
+        return sendResponseToClient(200, "Compressed File Uploaded to S3");
+    } catch (error) {
+        return sendResponseToClient(400, error);
```
Comment (mhazad) [inline fargate/compress.js:58]: Please explain with proper logic why do you think the error code should be 400?

**Review by mhazad — CHANGES_REQUESTED** [review]
Please address the feedback.

File: awsaf-serverless-engine/src/handler.js (hunk 1 above)
Comment (Awsaf780) [reply awsaf-serverless-engine/src/handler.js]: Resolved in commit - [f642467](https://github.com/CraftsmenLtd/coe-batch-q322/pull/345/commits/f64246799523b12299e1cfc721da8f26c88bbf08)
Mistake from my understanding of this concept. Will not happen again.

File: fargate/compress.js (hunk 3 above)
Comment (Awsaf780) [reply fargate/compress.js]: Resolved in commit - [f642467](https://github.com/CraftsmenLtd/coe-batch-q322/pull/345/commits/f64246799523b12299e1cfc721da8f26c88bbf08)

File: fargate/compress.js (hunk 4 above)
Comment (Awsaf780) [reply fargate/compress.js]: Resolved in commit - [f642467](https://github.com/CraftsmenLtd/coe-batch-q322/pull/345/commits/f64246799523b12299e1cfc721da8f26c88bbf08)

File: fargate/package.json (hunk 2 above)
Comment (Awsaf780) [reply fargate/package.json]: Resolved in commit - [12dd071](https://github.com/CraftsmenLtd/coe-batch-q322/pull/345/commits/12dd0711a529b7f8eea95a22452168cab28bda24)
Third-party library not needed. Node is enough.

**Review by mhazad — APPROVED** [review]
LGTM

**Pull request merged**